### PR TITLE
Reduce changes before warning to 1 and fix unwanted changes

### PR DESCRIPTION
**Pull request by ClementPasteau**
Also add warnings when animation is added or sprite is added

https://user-images.githubusercontent.com/4895034/220166930-ea36a67a-491c-44b3-98d1-c97a9f635f65.mov

### CODE REVIEW

File: newIDE/app/src/ObjectEditor/Editors/SpriteEditor/DirectionTools.js [hunk 1]
```diff
@@ -71,24 +72,32 @@ export default class DirectionTools extends Component<Props, State> {
...
-      timeBetweenFrames: formatTime(
-        this.props.direction.getTimeBetweenFrames()
-      ),
-      timeError: newTimeIsValid,
-    });
+    if (newTimeIsValid) {
+      direction.setTimeBetweenFrames(newTimeBetweenFrames);
+      this.setState({
+        timeBetweenFrames: formatTime(newTimeBetweenFrames),
+        timeError: newTimeIsValid,
```
Comment (AlexandreSi) [inline newIDE/app/src/ObjectEditor/Editors/SpriteEditor/DirectionTools.js:88]: I feel like this timeError is not used: if the time is valid, time error is set to true 😅

Comment (ClementPasteau) [reply newIDE/app/src/ObjectEditor/Editors/SpriteEditor/DirectionTools.js:88]: uh indeed

**Review by 4ian — COMMENTED** [review]
lgtm but maybe a second review would help to make sure there is not an obvious regression in the UI or a forgotten "this." somewhere?

File: newIDE/app/src/ObjectEditor/Editors/SpriteEditor/CollisionMasksEditor/index.js [hunk 2]
```diff
@@ -99,7 +99,12 @@ const CollisionMasksEditor = ({
       ? every(
           mapFor(0, spriteConfiguration.getAnimationsCount(), i => {
             const otherAnimation = spriteConfiguration.getAnimation(i);
-            return allSpritesHaveSameCollisionMasksAs(sprite, otherAnimation);
+            const same = allSpritesHaveSameCollisionMasksAs(
+              sprite,
+              otherAnimation
+            );
+            console.log(same);
```
Comment (AlexandreSi) [inline newIDE/app/src/ObjectEditor/Editors/SpriteEditor/CollisionMasksEditor/index.js:106]: To remove

File: newIDE/app/src/ObjectEditor/Editors/SpriteEditor/CollisionMasksEditor/index.js [hunk 3]
```diff
@@ -165,6 +162,11 @@ const CollisionMasksEditor = ({
     () => {
       if (!animation || !sprite) return;
 
+      console.log(
```
Comment (AlexandreSi) [inline newIDE/app/src/ObjectEditor/Editors/SpriteEditor/CollisionMasksEditor/index.js:165]: To remove

Comment (ClementPasteau) [reply newIDE/app/src/ObjectEditor/Editors/SpriteEditor/CollisionMasksEditor/index.js]: crap

**Review by AlexandreSi — APPROVED** [review]

**Pull request merged**